### PR TITLE
Clean up login client helpers

### PR DESCRIPTION
**Pull request by rnystrom**

### CODE REVIEW

File: Classes/Login/Client+AccessToken.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        completion: @escaping (Result<AccessTokenUser>) -> Void
+        ) {
+        send(GitHubAccessTokenRequest(
+            code: code,
+            clientId: Secrets.GitHub.clientId,
+            clientSecret: Secrets.GitHub.clientSecret)
+        ) { [weak self] result in
+            switch result {
+            case .success(let response):
+                self?.send(V3VerifyPersonalAccessTokenRequest(token: response.data.accessToken)) { result in
```
Comment (BasThomas) [inline Classes/Login/Client+AccessToken.swift]: Nit: Maybe put this in its own function?

File: Classes/Login/LoginSplashViewController.swift [hunk 2]
```diff
@@ -144,8 +143,13 @@ final class LoginSplashViewController: UIViewController, GithubSessionListener {
         safariController?.dismiss(animated: trueUnlessReduceMotionEnabled)
         state = .fetchingToken
 
-        client.requestAccessToken(code: code) { result in
-            self.handle(result: result, authMethod: .oauth)
+        client.requestAccessToken(code: code) { [weak self] result in
+            switch result {
+            case .error:
```
Comment (BasThomas) [inline Classes/Login/LoginSplashViewController.swift]: 🤔, why is this `.error` and not `.failure`?

Comment (BasThomas) [reply Classes/Login/LoginSplashViewController.swift]: Also, why do you handle the `.error` here first, when in other places you handle `.success` first?

Comment (rnystrom) [reply Classes/Login/LoginSplashViewController.swift]: Two different `Result`s and the naming diverged

**Pull request merged**